backend/internal/database/book.go

```diff
@@ -246,3 +254,90 @@ func (db *BookDatabase) UpdateBookDetails(ctx context.Context, id string,
 	return updatedBookDetails, err
 
 }
+
+/*
+ * Increments current day's clicks count in the book_analytics table given a book id
+ */
+ func (db *BookDatabase) UpdateBookAnalytics(ctx context.Context, id string) error {
+	var lastUpdated time.Time
+	var time = time.Now()
+	var dayInYear = time.YearDay()
+	var query string = "SELECT last_updated FROM book_analytics WHERE id = $1"
+
+	err := db.Conn.QueryRow(ctx, query, id).Scan(&lastUpdated)
+
+	if err != nil {
+		fmt.Print(err)
+		return errors.Wrap(err, "error on accessing book_analytics")
+	}
+
+	if(lastUpdated.YearDay() == time.YearDay() && lastUpdated.Year() == time.Year()) {
+		query = "UPDATE book_analytics SET " +
+		"clicks[$1] = COALESCE(clicks[$1], 0) + 1," +
+		"last_updated = $2 WHERE id = $3"
+	} else {
+		query = "UPDATE book_analytics SET " +
+		"clicks[$1] = 1," +
+		"last_updated = $2 WHERE id = $3"
+	}
+
+	_, err = db.Conn.Exec(ctx, query, dayInYear, time, id)
+
+	if err != nil {
+		fmt.Print(err)
+		return errors.Wrap(err, "error on updating book_analytics")
+	}
+
+	return nil
+
+}
+
+/*
+ * Returns an array containing a book's daily click counts given a range from the current day
+ */
+ func (db *BookDatabase) FetchBookAnalytics(ctx context.Context, id string, dRange int) ([]int, error) {
+	var analytics []int
+	var currTime = time.Now()
+	var dayInYear = currTime.YearDay()
+	var query string
+
+	var arg1 int
+	var arg2  int
+	var arg3 int
+
+	if dRange > dayInYear {
+		var prevYear = currTime.Year() - 1
+		var lastDay = time.Date(prevYear, time.December, 31, 0, 0, 0, 0, time.Local).YearDay()
+		arg1 = lastDay - (dRange - dayInYear) + 1
+		arg2 = lastDay
+		arg3 = dayInYear
+	} else {
+		arg1 = dayInYear - dRange + 1
+		arg2 = dayInYear
+		arg3 = 0
+	}
```

Comment:
Definitely the most annoying part of this API :P Can you please:

- Rename arg1/2/3 to something a little more clear? (Like `prevYearRangeStart` or something like that?)
- Add some comments explaining how this all works
- Maybe a couple of examples in comments on how this works with different date ranges?